File: app/src/main/java/com/hfad/pet_scheduling/data/local/entities/CompletedTask.kt
package com.hfad.pet_scheduling.data.local.entities

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "completed_tasks",
    foreignKeys = [
        ForeignKey(
            entity = ScheduleTask::class,
            parentColumns = arrayOf("taskId"),
            childColumns = arrayOf("taskId"),
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["taskId"]), Index(value = ["completedAt"])]
)
data class CompletedTask(
    @PrimaryKey
    val completedTaskId: String = java.util.UUID.randomUUID().toString(),

    val taskId: String,
    val completedAt: Long = System.currentTimeMillis(),
    val completedByUserId: String,
    val notes: String?,
    val scheduledTime: Long? // The original scheduled time this completion corresponds to
)

File: PetSchedulingiOS/PetSchedulingiOS/Models/CompletedTask.swift
//
//  CompletedTask.swift
//  PetSchedulingiOS
//

import Foundation

struct CompletedTask: Identifiable, Codable {
    var id: String { completedTaskId }
    let completedTaskId: String
    let taskId: String
    let completedAt: Int64
    let completedByUserId: String
    let notes: String?
    let scheduledTime: Int64?
}
